### PR TITLE
[XLSX] Add mergeCells reader

### PR DESCRIPTION
**Pull request by karakum**
Add possibility to read MergeCells metadata from XLSX sheet data xml.

### CODE REVIEW

**Comment by karakum** [comment]
@Slamdunk please merge PR

File: src/Reader/XLSX/SheetMergeCellsReader.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenSpout\Reader\XLSX;
+
+use OpenSpout\Common\Exception\IOException;
+use OpenSpout\Reader\Common\XMLProcessor;
+use OpenSpout\Reader\Wrapper\XMLReader;
+
```
Comment (Slamdunk) [inline src/Reader/XLSX/SheetMergeCellsReader.php:10]:
```suggestion

/**
 * @internal
 */
```

File: src/Reader/XLSX/SheetMergeCellsReader.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     */
+    public function __construct(
+        string $filePath,
+        string $sheetDataXMLFilePath,
+        XMLReader $xmlReader,
+        XMLProcessor $xmlProcessor
+    ) {
+        $this->filePath = $filePath;
+        $this->sheetDataXMLFilePath = $this->normalizeSheetDataXMLFilePath($sheetDataXMLFilePath);
+        $this->xmlReader = $xmlReader;
```
Comment (Slamdunk) [inline src/Reader/XLSX/SheetMergeCellsReader.php]: What's the point of requesting and saving the `$xmlReader` if it's already injected in the `$xmlProcessor`?

File: src/Reader/XLSX/SheetMergeCellsReader.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+        // Register all callbacks to process different nodes when reading the XML file
+        $this->xmlProcessor = $xmlProcessor;
+        $this->xmlProcessor->registerCallback(self::XML_NODE_MERGE_CELL, XMLProcessor::NODE_TYPE_START, [$this, 'processMergeCellsStartingNode']);
```
Comment (Slamdunk) [inline src/Reader/XLSX/SheetMergeCellsReader.php]: We don't need a class-alike callback, a simple local callback is enough, so we can also make the `$mergeCells` property `readonly`

File: src/Reader/XLSX/SheetMergeCellsReader.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+        // @infection-ignore-all
```
Comment (Slamdunk) [inline src/Reader/XLSX/SheetMergeCellsReader.php]: Please remove all the Infection related annotations, I want to see the errors every time

**Review by Slamdunk — CHANGES_REQUESTED** [review]
Hi, it's a very good start, few twaks and we can merge this :muscle:

File: src/Reader/XLSX/SheetMergeCellsReader.php (hunk 2 above)
Comment (karakum) [reply src/Reader/XLSX/SheetMergeCellsReader.php]: @Slamdunk  I don't know why... I just make like in SheetHeaderReader

Comment (karakum) [reply src/Reader/XLSX/SheetMergeCellsReader.php]: @Slamdunk I guess for `$this->xmlReader->openFileInZip(...)`

File: src/Reader/XLSX/SheetMergeCellsReader.php (hunk 3 above)
Comment (karakum) [reply src/Reader/XLSX/SheetMergeCellsReader.php]: @Slamdunk it is impossible due to XMLProcessor require callback as array of object and method name... (see XMLProcessor.php:(102))

**Review by Slamdunk — APPROVED** [review]

**Comment by Slamdunk** [comment]
Thank you

**Pull request merged**